src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePluginTest.java

```diff
@@ -97,761 +97,441 @@
 	}
 
-	// test case: When calling the updateHardwareRequirements method with a set of
-	// outdated flavors, a collection of valid resources must be obtained in the
-	// cloud and compared to existing flavors, adding new flavors to the set when
-	// they do not exist.
+	// test case: when invoking the requestInstance method with valid compute order and
+	// cloud user, the plugin should mount an appropriate ONe create compute request,
+	// run it via the ONe client and return the instance id.
 	@Test
-	public void testUpdateHardwareRequirementsSuccessfully() throws UnexpectedException {
+	public void testRequestInstance() throws FogbowException {
 		// set up
-		ImagePool imagePool = Mockito.mock(ImagePool.class);
-		PowerMockito.mockStatic(OpenNebulaClientUtil.class);
-		PowerMockito.when(OpenNebulaClientUtil.getImagePool(Mockito.any(Client.class))).thenReturn(imagePool);
-
-		Image image = Mockito.mock(Image.class);
-		Iterator<Image> imageIterator = Mockito.mock(Iterator.class);
-		Mockito.when(imageIterator.hasNext()).thenReturn(true, false);
-		Mockito.when(imageIterator.next()).thenReturn(image);
-		Mockito.when(imagePool.iterator()).thenReturn(imageIterator);
-		Mockito.when(image.xpath(IMAGE_SIZE_PATH)).thenReturn(IMAGE_SIZE_STRING_VALUE);
-
-		TemplatePool templatePool = Mockito.mock(TemplatePool.class);
-		PowerMockito.when(OpenNebulaClientUtil.getTemplatePool(Mockito.any(Client.class))).thenReturn(templatePool);
+		CreateComputeRequest createComputeRequest = this.getCreateComputeRequest();
 
-		Template template = Mockito.mock(Template.class);
-		Iterator<Template> templateIterator = Mockito.mock(Iterator.class);
-		Mockito.when(templateIterator.hasNext()).thenReturn(true, false);
-		Mockito.when(templateIterator.next()).thenReturn(template);
-		Mockito.when(templatePool.iterator()).thenReturn(templateIterator);
-		Mockito.when(template.getId()).thenReturn(ONE_STRING_VALUE);
-		Mockito.when(template.getName()).thenReturn(FAKE_NAME);
-		Mockito.when(template.xpath(TEMPLATE_CPU_PATH)).thenReturn(ONE_STRING_VALUE);
-		Mockito.when(template.xpath(TEMPLATE_MEMORY_PATH)).thenReturn(MEMORY_STRING_VALUE);
-		Mockito.when(template.xpath(TEMPLATE_IMAGE_ID_PATH)).thenReturn(ONE_STRING_VALUE);
-
-		Mockito.doReturn(DISK_VALUE_8GB).when(this.plugin).getDiskSizeFromImageSizeMap(Mockito.anyMap(), Mockito.anyString());
-
-		int cpu = CPU_VALUE_1;
-		int memory = MEMORY_VALUE_1024;
-		int disk = DISK_VALUE_8GB;
-		String flavorId = ONE_STRING_VALUE;
-		String name = FAKE_NAME;
-		HardwareRequirements flavor = new HardwareRequirements(name, flavorId, cpu, memory, disk);
-		TreeSet<HardwareRequirements> expected = new TreeSet<HardwareRequirements>();
-		expected.add(flavor);
-
-		Client client = Mockito.mock(Client.class);
+		Mockito.doReturn(createComputeRequest).when(this.plugin).getCreateComputeRequest(
+				Mockito.any(Client.class), Mockito.any(ComputeOrder.class));
+		Mockito.doReturn(FAKE_ID).when(this.plugin).doRequestInstance(
```

Comment:
I think the **synchronized** block of this method in line 84 of the plug-in should be extracted to a method and called after the _doquestInstance_ method, because this data should only be set after the success of the order.

Reply:
done